### PR TITLE
[IAP] M2 expandable card feature details

### LINKED ISSUE #10199: [IAP] M2 plans expandable card main features
As continuation of https://github.com/woocommerce/woocommerce-ios/issues/10189 , we need to implement the expandable/collapsable feature details for each card when we tap on the "View Plan Features" button

Screenshot 2023-07-10 at 20 17 01

| Essential | Performance |
|--------|--------|
| Screenshot 2023-07-10 at 20 19 27 | Screenshot 2023-07-10 at 20 20 08 |

### PR DESCRIPTION
**Pull request by iamgabrielma**
Closes: #10199 

## Description
This PR adds the list of main features for Essential and Performance plans to the existing expandable card. For this, we start by renaming the previous `WooPlan` to `LegacyWooPlan`, since this struct currently contains a lot of unnecessary information for M2, and create a new WooPlan with the minimum necessary.

## Changes
- Renames  `WooPlan` to `LegacyWooPlan`: As with M1, the plan features data here is hardcoded since we plan to load this dynamically via the API in future milestones (~M4), and remove all legacy classes that we no longer need.
- Creates a new `WooPlanCardFeaturesView`, which displays the features of each plan

## Testing instructions

On a Free Trial site, eligible for IAP (a new site can be created in Menu > Switch Sites > Add New Site) tap on Purchase Now and observe that:
- Tapping on `View Essential features` shows the list of features in the screenshot below
- Tapping on `View Performance features` shows the text "_Everything in essential, plus:_" at the top, plus a different set of features (screenshot below).

| Essential features | Performance features |
|--------|--------|
| Simulator Screen Shot - iPhone 13 Pro - 2023-07-12 at 15 34 33 | Simulator Screen Shot - iPhone 13 Pro - 2023-07-12 at 15 34 41  |

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Upgrades/WooPlanCardView.swift
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Upgrades/WooPlanCardView.swift]: Oops I forgot the comment on these, will add them before merging.

File: WooCommerce/Classes/ViewRelated/Upgrades/WooPlanCardView.swift
Comment (toupper) [inline WooCommerce/Classes/ViewRelated/Upgrades/WooPlanCardView.swift]: Do we have a view model for this? It seems like all this logic belongs there, and can be tested as well.

File: WooCommerce/Classes/ViewRelated/Upgrades/WooPlanCardView.swift
Comment (toupper) [inline WooCommerce/Classes/ViewRelated/Upgrades/WooPlanCardView.swift]: Is this popular plans logic used elsewhere? It seems like deciding which plans are popular needs to be in a more accessible/central place, such as the `InAppPurchasesForWPComPlansManager` class

File: Yosemite/Yosemite/Model/WooPlans/LegacyWooPlan.swift
Comment (toupper) [inline Yosemite/Yosemite/Model/WooPlans/LegacyWooPlan.swift]: Not a blocker, but here and in other places in this PR I see there's an empty line after the function/class definition, but AFAIK we have the convention of not leaving that :)

**Comment by toupper** [comment]
Thanks for the PR @iamgabrielma! I add here a few remarks and questions:

When testing, I see the price is different than what you have in your screenshots:




Apart from that, I have some concerns about the name change to `LegacyWooPlan`. The naming might be confusing, as my first thought was that that plan is an old IAP product that we have to handle, but I think the intention behind that is different. If that plan contains a lot of unnecessary information, can we just remove it, or discern the new code we need with the feature flag? That way we don't merge code in `trunk` that's not used, or duplicated. 
On the other hand, I see that this approach has been used before with other elements (e.g. `LegacyUpgradesViewModel` v `UpgradesViewModel`) so it might be too late to change it.

File: WooCommerce/Classes/ViewRelated/Upgrades/WooPlanCardView.swift
Comment (iamgabrielma) [reply WooCommerce/Classes/ViewRelated/Upgrades/WooPlanCardView.swift]: Fixed here: 8fc7efde9966b8bffd139474096d417a1197988e

File: WooCommerce/Classes/ViewRelated/Upgrades/WooPlanCardView.swift
Comment (iamgabrielma) [reply WooCommerce/Classes/ViewRelated/Upgrades/WooPlanCardView.swift]: Good point, I renamed the property to `shouldDisplayIsPopularBadge` and moved it to the ViewModel here: 2fe488acd833e42ecc880d72f6057e735a865612

File: WooCommerce/Classes/ViewRelated/Upgrades/WooPlanCardView.swift
Comment (iamgabrielma) [reply WooCommerce/Classes/ViewRelated/Upgrades/WooPlanCardView.swift]: It is not used anywhere else. Since we only use it for that specific view due to design requirements I just moved it in the ViewModel that drives that view's data (https://github.com/woocommerce/woocommerce-ios/commit/2fe488acd833e42ecc880d72f6057e735a865612)

File: Yosemite/Yosemite/Model/WooPlans/LegacyWooPlan.swift
Comment (iamgabrielma) [reply Yosemite/Yosemite/Model/WooPlans/LegacyWooPlan.swift]: Good catch! Removed whitespace in different places with 709d7cea3c21f5ae9403212ee9021f151cbee1e7

**Comment by iamgabrielma** [comment]
Thanks for the review! 

> When testing, I see the price is different than what you have in your screenshots:

That's right, I didn't realise I was using a non-supported site for testing when I made the screenshots, sorry for that! Since we only support USD purchases, the prices you see should be these:

| Annually | Monthly |
|--------|--------|
| Simulator Screen Shot - iPhone 11 Pro - 2023-07-13 at 10 40 57 | Simulator Screen Shot - iPhone 11 Pro - 2023-07-13 at 10 41 00 | 

This PR doesn't modify these though, so we're good!

**Comment by iamgabrielma** [comment]
> Apart from that, I have some concerns about the name change to LegacyWooPlan. The naming might be confusing, as my first thought was that that plan is an old IAP product that we have to handle, but I think the intention behind that is different. If that plan contains a lot of unnecessary information, can we just remove it, or discern the new code we need with the feature flag? That way we don't merge code in trunk that's not used, or duplicated.
On the other hand, I see that this approach has been used before with other elements (e.g. LegacyUpgradesViewModel v UpgradesViewModel) so it might be too late to change it.

That's a good point, and definitely agree to keep the codebase as clean as possible. 

At the moment we opted for duplicating and naming these as `Legacy_` as an intermediate strategy since we have to gradually replace parts of the system due to different constraints, mainly M1 and M2 available plans and capabilities being different (not only which plans are available, but also the posibility of changing between them), but also the API development on MobilePay is still ongoing.

Added to this, we had to hardcode a lot of data to match the initial designs, since we don't have an endpoint that gives us this informations, and the designs for all of this have completely changed with M2, creating a need for simpler components while we still keep the old ones.

Due to these constraints we opted for this approach, so we had both a safer roll-out of the feature (specially since they need to approve both the IAP products and the app in separate instances), as well as a fallback to default to if something went wrong with M2.

We explored different options and decided that duplicating the logic would be the best path forward for the time being, either until we can release M2 or perhaps until we can just fetch this data from the API on M4. This approach was discussed on a call, but there's some reference here: p1688741478426109?thread_ts=1688722642.908989&cid=C025A8VV728-slack-C025A8VV728

**Comment by toupper** [comment]
Thanks for your answer and for providing more context @iamgabrielma!

Even with that, I see that we're duplicating logic not related to the additions of M2, which makes it difficult to maintain and understand. Also, with this approach, we take the risk of leaving it duplicated indefinitely, if for whatever reason we stop the development of this project. I'm wondering if using a feature flag to add the M2 features would accomplish what we want (new M2 features, safe roll-out) without the caveats stated above. That would also improve the design, as we would be forced to decouple the new logic from the one that should stay intact.

I see that we already moved in this direction and it's difficult to change now, but it's worth taking that into consideration for future decisions.

File: Yosemite/Yosemite/Model/WooPlans/LegacyWooPlan.swift
Comment (toupper) [inline Yosemite/Yosemite/Model/WooPlans/LegacyWooPlan.swift]: I see now that we changed the name of the file to `LegacyWooPlan`, while keeping `WooPlan` in that file. Can we move LegacyWooPlan to its own file?

**Review by toupper — APPROVED** [review]
LGTM 🚢 :shipit:

File: Yosemite/Yosemite/Model/WooPlans/LegacyWooPlan.swift
Comment (iamgabrielma) [reply Yosemite/Yosemite/Model/WooPlans/LegacyWooPlan.swift]: Certainly. Changed on https://github.com/woocommerce/woocommerce-ios/pull/10219/commits/13f3798c36c82342b16ea79ddcc6337962cde04c

**Pull request merged**
